### PR TITLE
Added UserMulti Threading

### PR DESCRIPTION
**Pull request by ryanschubert99**
Added User multi threading for a given set of user input and output config values, also fixed some of the test harness but its still not passing.

### CODE REVIEW

File: src/src/ComputationCoordinatorImp.java [hunk 1]
```diff
@@ -1,52 +1,74 @@
...
+          // Generate the unique output file name for this thread
+          String uniqueOutputFileName = generateUniqueFileName(outputFileName, threadIndex);
+          ComputeEngineImp computeEng = new ComputeEngineImp(dataStorage);
+
+          // Call the method to perform the computation with the unique output file name
+          computeEng.multiplyMatrix(dataStorage.getMatrices());
+          dataStorage.writeOutput(uniqueOutputFileName, ";"); // Write output to the unique file
+
+        } catch (Exception e) {
+          e.printStackTrace();
```
Comment (LourisA11) [inline src/src/ComputationCoordinatorImp.java:50]: this does not need to throw an exception, istead it should be ComputeResultImpl(ComputeResultStatus.FAILURE, "An error occurred during computation.");

**Review by LourisA11 — COMMENTED** [review]
See if there is a way to move matrixprint files into .gitignore

File: test/tests/TestMultiUser.java [hunk 2]
```diff
@@ -20,23 +20,34 @@ public class TestMultiUser {
...
+      // Run the test user
       singleThreadedOut.deleteOnExit();
       testUsers.get(i).run(singleThreadedOut.getCanonicalPath());
+
+      // Check if the file exists
+      if (singleThreadedOut.exists()) {
+        System.out.println("Single-threaded temp file created: " + singleThreadedOut.getAbsolutePath());
+      } else {
+        System.out.println("Single-threaded temp file not found: " + singleThreadedOut.getAbsolutePath());
+      }
```
Comment (LourisA11) [inline test/tests/TestMultiUser.java:50]: I like that you added this in! good thinking!

File: test/tests/TestUser.java [hunk 3]
```diff
@@ -11,10 +15,44 @@ public TestUser(ComputationCoordinatorImp coordinator) {
...
+        System.out.println("Input Type: " + inputType);
+        System.out.println("Output File: " + inputFile);
+        System.out.println("Number to Generate: " + numGenerate);
+        System.out.println("Number of Rows: " + numRows);
+        System.out.println("Number of Columns: " + numCols);
+        System.out.println("Output Type: " + outputType);
+        System.out.println("Output File Name: " + outputFileName);
+        System.out.println("Out or Compute: " + outOrCompute);
+
+        // Call the computation method with parsed values
```
Comment (LourisA11) [inline test/tests/TestUser.java:50]: i like that you added a file reader, all though i believe this should have been handled by DS

**Review by LourisA11 — APPROVED** [review]

File: test/tests/TestMultiUser.java (hunk 2 above)
Comment (ryanschubert99) [reply test/tests/TestMultiUser.java:50]: Yeah the test wasnt passing, and printing this out helped debug the test.

File: test/tests/TestUser.java (hunk 3 above)
Comment (ryanschubert99) [reply test/tests/TestUser.java:50]: Yeah i agree, we will implement the csv file config in the main code in a future pr.

**Pull request merged**